### PR TITLE
Add linux x86 build CI for pull request

### PR DESCRIPTION
**Pull request by clamp03**

### CODE REVIEW

**Review by gbalykov — APPROVED** [review]
Ah, I see that CI uses `build-runtime.sh` script directly for runtime build instead of `build.sh`. Looks good.

**Review by safern — APPROVED** [review]
Pipeline changes LGTM.

File: src/coreclr/vm/CMakeLists.txt [hunk 1]
```diff
@@ -952,6 +952,7 @@ add_library_clr(cee_dac ${VM_SOURCES_DAC})
 add_dependencies(cee_dac eventing_headers)
 set_target_properties(cee_dac PROPERTIES DAC_COMPONENT TRUE)
 target_precompile_headers(cee_dac PRIVATE [["common.h"]])
+target_compile_definitions(cee_dac PUBLIC CORECLR_EMBEDDED)
```
[the remark this answers is not in the document]
Comment (VSadov) [reply src/coreclr/vm/CMakeLists.txt:955]: @clamp03 - yes on Windows compiler could be more tolerant to symbol duplication. 
Did you get errors when you tried  the change in https://github.com/dotnet/runtime/pull/60817/commits/4147fb6fdb3df8c92522815455c92e2185e13282 ?

[the remark this answers is not in the document]
Comment (clamp03) [reply src/coreclr/vm/CMakeLists.txt]: I didn't get any errors. Reverted the last commit. Now it includes ```gcdecode.cpp``` in only windows. Thank you!

**Comment by clamp03** [comment]
@jkotas @VSadov Could you please review again and then merge if it is good? Thank you.

**Review by jkotas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**